### PR TITLE
Update `Table.join` defaults and change unique name suffix from `_n` to ` n`

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

A few small changes, lots of lines because this affected lots of tests:
- `Table.join` now defaults to `Join_Kind.Left_Outer`, to avoid losing rows in the left table unexpectedly. If the user really wants to have an Inner join, they can switch to it.
- `Table.join` now defaults to joining columns by name not by index - it looks in the right table for a column with the same name as the first column in left table.
- Missing Input Column errors now specify which table they refer to in the join.
- The unique name suffix in column renaming / default column names when loading from file is now a space instead of underscore.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by jdunkerley — APPROVED** [review]
LGTM

File: distribution/lib/Standard/Table/0.0.0-dev/src/Errors.enso [hunk 1]
```diff
@@ -12,14 +12,17 @@ polyglot java import org.enso.table.error.EmptySheetException
...
     ## PRIVATE
 
        Convert a missing input error to a human-readable form.
     to_display_text : Text
     to_display_text self =
-        "The criteria "+self.criteria.to_text+" did not match any columns."
+        where = case self.where of
+            Nothing -> "."
+            location : Text -> " in "+location+"."
+        "The criteria "+self.criteria.to_text+" did not match any columns"+where
```
Comment (GregoryTravis) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Errors.enso:25]: Need a space after "columns".

File: distribution/lib/Standard/Table/0.0.0-dev/src/Errors.enso [hunk 2]
```diff
@@ -12,14 +12,17 @@ polyglot java import org.enso.table.error.EmptySheetException
 type Missing_Input_Columns
     ## PRIVATE
        One or more columns not found in the input table.
-    Error (criteria : [Text])
+    Error (criteria : [Text]) (where:Text|Nothing = Nothing)
```
Comment (GregoryTravis) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Errors.enso:15]: What is meant to go in 'where'?

**Review by GregoryTravis — APPROVED** [review]

File: distribution/lib/Standard/Table/0.0.0-dev/src/Errors.enso (hunk 1 above)
Comment (radeusgd) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Errors.enso:25]: No, if `where` is `.` the space should not be there, and in the other branch the space is added to `where`.

File: distribution/lib/Standard/Table/0.0.0-dev/src/Errors.enso (hunk 2 above)
Comment (radeusgd) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Errors.enso:15]: A location description, dependent on the particular operation. I will add a comment in subsequent PR.

**Pull request merged**